src/index.test.ts

```diff
@@ -441,10 +441,7 @@ testLog(
       request: {
         _options: {
           method: 'get',
-          headers: {
-            Accept: 'application/json, text/plain, */*',
-            authorization: '[Redacted]',
-          },
```

Comment:
This is now truncated as a result of fixing the `max object depth`.

Reply:
Random stream of consciousness thought, do we know if the trimming impacts a potentially deep `err.stack` trace? Perhaps we could pull some (non-sensitive) ones out of Splunk and have a test case around it? The stack trace tends to be the highest value field in an error log, so retaining a long stack may be a safer default.

Reply:
I've excluded the `stack` field from being trimmed, though the sharing of a trimmer between serializers means that its not just the top level fields of `err` and `errWithCause` that are affected by this exclusion.

Reply:
I was poking around with this, do you have thoughts on 7b1f2a1dc05f91644b348f316bc9c417cc472589?

Reply:
Sure do that's way cleaner I like it!!
Cleaned up and condensed all the bits of spread out logic that I implemented and didn't like 😆 🙏

Reply:
Thanks for the change, refactoring is the easy part 😉